### PR TITLE
Convert ./components/widgets/loading/loading_wrapper.tsx from Class C…

### LINKED ISSUE #24761: Convert ./components/widgets/loading/loading_wrapper.tsx from Class Component to Function Component
We want to move most of our codebase from Class Components to Function Components. This decision is based on [React recommendations](https://react.dev/reference/react/Component).

The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring.

Here are a few examples of the required changes:
https://github.com/mattermost/mattermost/pull/24187
https://github.com/mattermost/mattermost/pull/24210
https://github.com/mattermost/mattermost/pull/24183
https://github.com/mattermost/mattermost/pull/24181

If the tests do not work because they rely on the behavior of Class Components they also have to be migrated to functionaly test the same behavior. We prefer the use of `react-testing-library` for new tests, but it is not required.

When submitting the PR, please make sure to identify where and when the component shows, for QA to properly test the changes.

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA LINK: MM-54797

### PR DESCRIPTION
**Pull request by sjcode99**
This PR convert ./components/widgets/loading/loading_wrapper.tsx from Class Component to Function Component.

Fixes #24761

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/widgets/loading/loading_wrapper.test.tsx [hunk 1]
```diff
@@ -43,15 +43,15 @@ describe('components/widgets/loading/LoadingWrapper', () => {
...
+            const wrapper = mount(
                 <LoadingWrapper
                     loading={testCase.loading}
                     text={testCase.text}
                 >
                     {testCase.children}
                 </LoadingWrapper>,
             );
-            expect(wrapper).toMatchInlineSnapshot(testCase.snapshot);
+            expect(wrapper).toMatchSnapshot(testCase.snapshot);
```
Comment (larkox) [inline webapp/channels/src/components/widgets/loading/loading_wrapper.test.tsx:54]: Not really opposed but why these changes?

That being said, if we do it like this, we can remove the `testCase.snapshot` from here, and all the snapshot in the test cases in the file.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Looks good to me, just one question, that may lead to some changes.

File: webapp/channels/src/components/widgets/loading/loading_wrapper.tsx [hunk 2]
```diff
@@ -6,24 +6,17 @@ import React from 'react';
 import LoadingSpinner from './loading_spinner';
 
 type Props = {
-    loading: boolean;
-    text: React.ReactNode;
-    children: React.ReactNode;
+    loading?: boolean;
```
Comment (devinbinnie) [inline webapp/channels/src/components/widgets/loading/loading_wrapper.tsx:9]: I don't think the optional arguments are necessary here if we're supporting default properties. `ReactNode` includes `null` in the union type and `booleans` are always truthy/falsy.

**Review by devinbinnie — CHANGES_REQUESTED** [review]
LGTM other than the one comment I have. Thanks @sjcode99!

File: webapp/channels/src/components/widgets/loading/loading_wrapper.tsx (hunk 2 above)
Comment (devinbinnie) [reply webapp/channels/src/components/widgets/loading/loading_wrapper.tsx:9]: Gotcha. I feel like this is something that could be improved with functional components, but I guess we make trade-offs for performance/other forms of simplicity at the expense of type correctness.

File: webapp/channels/src/components/widgets/loading/loading_wrapper.test.tsx (hunk 1 above)
Comment (sjcode99) [reply webapp/channels/src/components/widgets/loading/loading_wrapper.test.tsx:54]: yes we can do that too. I changed from `toMatchInlineSnapshot` to  `toMatchSnapshot` because when the data passed was empty then the functional component returned `null` whereas  snapshot expected `"''"`. and this was giving error.

Please suggest

Comment (larkox) [reply webapp/channels/src/components/widgets/loading/loading_wrapper.test.tsx:54]: Looks good to me. Remove the inline snapshots from the testcase (since they are not used anymore) and that will be all :smile:

Comment (sjcode99) [reply webapp/channels/src/components/widgets/loading/loading_wrapper.test.tsx]: updated the PR with the required changes. Please have a look.

**Review by larkox — APPROVED** [review]

**Comment by syedzubeen** [comment]
@lindy65 I would like to test this change as QA.

**Comment by syedzubeen** [comment]
Yup, I have an acc. on the Comm Server.

**Comment by lindy65** [comment]
Thanks @syedzubeen - what is your username on `community`?

**Comment by syedzubeen** [comment]
> Thanks @syedzubeen - what is your username on `community`?

@zubeen

**Comment by syedzubeen** [comment]
@lindy65 
I'm a little lost here. I see changes have been made to a lot of files in this PR and I don't know where I should start looking from. 
Is this correct ?

mattermost

**Comment by syedzubeen** [comment]
@larkox  Got it. And do I add GIFs to this PR documenting the places that I have tested so far?

**Comment by lindy65** [comment]
Hi @syedzubeen - no need to add GIFs for all areas you've tested the SAVE button loading wrapper.  One or two will suffice.

**Comment by syedzubeen** [comment]
Done! 
Let me know if there's something I left out.

mattermost-QA
